weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -648,6 +655,40 @@ def objs_query(self, req: tsi.ObjQueryReq) -> tsi.ObjQueryRes:
 
         return tsi.ObjQueryRes(objs=[_ch_obj_to_obj_schema(obj) for obj in objs])
 
+    def obj_delete(self, req: tsi.ObjDeleteReq) -> tsi.ObjDeleteRes:
+        # 1. Read the entire object from the db based on id and digest
+        # 2. Set deleted_at to the current time
+        # 3. Insert this object into object_versions
+        #    - This object should be IDENTICAL to the original, including
+        #      the created_at time, this will become the only copy of the
+        #      object when the db deduplicates on primary key
+        db_obj = self.obj_read(
+            tsi.ObjReadReq(
+                project_id=req.project_id,
+                object_id=req.object_id,
+                digest=req.digest,
+            )
+        ).obj
+
+        deleted_at = datetime.datetime.now()
+        ch_obj = ObjDeleteCHInsertable(
+            project_id=req.project_id,
+            object_id=req.object_id,
+            digest=req.digest,
+            kind=get_kind(db_obj.val),
+            val_dump=json.dumps(db_obj.val),
```

Comment:
interestingly, if we made this value something like `DELETED`, then we would never need to clean this up

Reply:
oh nice! we could add a flag here for "hard delete" which sets that